[TC-9065] Update webhook connectors (#5)
tradecloud/tradecloud-docs-connectors: webhook-connector/configure-the-webhook.md

# Configure the webhook

> ## Configure your webhooks in the Tradecloud1 portal

Reviewer: Nit; we can remove this header. Now we have 3 header lines saying the same thing..

<img width="861" alt="Screenshot 2023-11-15 at 10 20 27" src="https://github.com/tradecloud/tradecloud-docs-connectors/assets/46396634/8cf7af20-1336-4c71-988e-d7a59ccf1455">

Author: First header `Configure the webhook` removed

### Select one or more webhooks

> As a company admin configure one or more webhooks in your company settings on the [Tradecloud1 platform](http://portal.tradecloud1.com):

Reviewer: ```suggestion
A company admin can configure one or more webhooks in your company settings on the [Tradecloud1 platform](http://portal.tradecloud1.com):
```

![Webhook Settings](../.gitbook/assets/webhook-settings.png)

* Select **My company** in the menu below your avatar.

> * The **Settings** tab in the Company menu should be selected.

Reviewer: Why reword it? imo it's easier to scan this text if the bullets have the same sentence structure.

Author: If you select **My company** then **Settings** is already selected.

Reviewer: I see. fair enough

* Configure one or more **Webhook Integration**s below.

### Orders Webhook Integration

Enable the **Order Webhook Integration** if you want to receive a webhook trigger when an order has been issued or changed:

![Order Events](../.gitbook/assets/webhook-order-events.png)

> A default set of order events will be enabled. The actual set you want to select wil be dependent on the capabilities of your integration and ERP system. You can find a list of events here:

Reviewer: "dependent on the capabilities of your integration and ERP system" -> And your process flow / functional requirements?

Author: Added ` and your order process requirements.`

{% page-ref page="order-events.md" %}

> ![Order Delivery Schedule](../.gitbook/assets/webhook-order-delivery-schedule.png)

Reviewer: Nit; maybe add a small subheader here?

Author: Added

In case of the order webhook you can choose between 
* Multiple delivery lines per order line, also known as the **native** delivery schedule.

> * Only on delivery line per order line, also known as the **simple**  delivery schedule.

Reviewer: ```suggestion
You can configure the way the delivery schedule is included in the body of a webhook POST request. You can choose between:
* Multiple delivery lines per order line, also known as the **native** delivery schedule. This will result in an `orderEvent` in the POST request body.
* Only on delivery line per order line, also known as the **simple**  delivery schedule. This will result in a `simpleOrderEvent` in the POST request body.
```

Author: Updated

See [the API manual](https://docs.tradecloud1.com/api/introduction/api/delivery-schedule) to read about the native versus the simple delivery schedule.

### Order Documents Webhook Integration

Enable the **Order Documents Webhook Integration** if you want to receive a webhook trigger when an order document has been attached:

![Order Documents Events](../.gitbook/assets/webhook-order-documents-events.png)

> A default set of order document events will be enabled. The actual set you want to select wil be dependent on the capabilities of your integration and ERP system. You can find a list of events here:

Reviewer: ```suggestion
You may choose whether you want to receive events for documents added by the buyer of an order, the suppplier, or both. You can find a list of events here:
```

{% page-ref page="order-documents-events.md" %}

### Shipments Webhook Integration

> Enable the **Order Webhook Integration** if you want to receive a webhook trigger when a shipment has been issued or changed:

Reviewer: ```suggestion
Enable the **Shipment Webhook Integration** if you want to receive a webhook trigger when a shipment has been issued or changed:
```

![Shipment Events](../.gitbook/assets/webhook-shipment-events.png)

> A default set of shipments events will be enabled. The actual set you want to select wil be dependent on the capabilities of your integration and ERP system. You can find a list of events here:

Reviewer: Same here; also depends on your functional requirements etc?

Author: Added

{% page-ref page="shipment-events.md" %}

> ### Configure method, url and credentials

Reviewer: ```suggestion
### Method, URL and credentials
```

> ![Order Webhook Congfig](../.gitbook/assets/webhook-order-config.png)

Reviewer: Also render the other configuration panels side-by-side to avoid confusion? As it's merely an example..

Author: Not sure what you suggest.
The images are focussed and examples indeed.

Reviewer: I meant using 3 images in 3 columns; for the order webhook settings, order document settings and shipment webhook settings. But it's a nit..

Author: I think its clear like this.

> * Select **GET** or **POST** as HTTP method.

Reviewer: ```suggestion
Finally, configure the webhook method, URL and credentials
* Select **GET** or **POST** as HTTP method.
```

* Enter your webhook **URL**

>   * **https** is required.

Reviewer: ```suggestion
  * Must start with **https://**
```

>   * use the **{orderId}** variable in the URL in case of **GET** method.

Reviewer: ```suggestion
  * You can use the **{orderId}** variable in the URL in case of **GET** method.
```

Author: Updated with `You must`

Reviewer: I don't think it's required, right?

Author: It must be used, else you do not know which order has been triggered and the webhook makes no sense.
